pkg/virtctl/imageupload/imageupload.go

```diff
@@ -667,3 +717,48 @@ func getUploadProxyURL(client cdiClientset.Interface) (string, error) {
 	}
 	return "", nil
 }
+
+// check is accessMode from StorageProfile is set and is correct (not ReadOnly),
+func profileSpecifiesCorrectAccessMode(client kubecli.KubevirtClient, storageClass string) (bool, error) {
+	if storageClass == "" {
+		return false, nil
+	}
+
+	storageProfile, err := client.CdiClient().CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClass, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return false, nil
+		}
+
+		return false, fmt.Errorf("accessMode not provided, cannot get storage class %s", err)
+	}
+
+	if len(storageProfile.Status.ClaimPropertySets) > 0 &&
+		len(storageProfile.Status.ClaimPropertySets[0].AccessModes) > 0 {
+		accessMode := storageProfile.Status.ClaimPropertySets[0].AccessModes[0]
+
+		if accessMode == v1.ReadOnlyMany {
+			return false,
+				fmt.Errorf("cannot upload to a readonly volume, use either ReadWriteOnce or ReadWriteMany if supported")
+		}
+		if accessMode == v1.ReadWriteOnce || accessMode == v1.ReadWriteMany {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func getDefaultStorageClassName(client kubecli.KubevirtClient) (string, error) {
+	storageClasses, err := client.StorageV1().StorageClasses().List(context.Background(), metav1.ListOptions{})
```

Comment:
How common is it that people have access to this global resource on limited accounts?

Reply:
good question, @awels do you have any answer to this question?

Reply:
Never, you cannot rely on storage classes being available to regular users.

Reply:
Does it mean we need to think about ways to improve the code here?

Reply:
Yes, remember virtctl is a client, just like kubectl. So we shouldn't have to figure out access modes and volume modes if not provided. We just fill in the parts of the DV storage section that we did get, and let the backend figure out the rest. This does mean we cannot give the nice error message about uploading to readonly volumes if the storage profile determines that is the access mode. But this should be a rare occurrence.

Reply:
so the current implementation is usable but no so good when the user tries to use default storageClass and does not specify AccessMode, I am on it again, to improve this

Reply:
oh no. good catch @rmohr 

@brybacki this actually causes a regression. Previously, storageClass wasn't required, and now leaving storageClass empty will cause the upload command to fail. It's also unclear if normal users have permission to read the CDI storage profiles as well. 

I think we need to consider reverting this change

Reply:
yes, I am on it, we have already discussed it on Friday with @awels